storagemarket/deal_execution.go

```diff
@@ -242,12 +252,30 @@ func (p *Provider) transferAndVerify(ctx context.Context, pub event.Emitter, dea
 	}
 
 	p.dealLogger.Infow(deal.DealUuid, "commP matched successfully: deal-data verified")
-	return p.updateCheckpoint(ctx, pub, deal, dealcheckpoints.Transferred)
+	return p.updateCheckpoint(pub, deal, dealcheckpoints.Transferred)
 }
 
 func (p *Provider) waitForTransferFinish(ctx context.Context, handler transport.Handler, pub event.Emitter, deal *types.ProviderDealState) error {
 	defer handler.Close()
 	defer p.transfers.complete(deal.DealUuid)
+	tlog := make(map[int]struct{})
```

Comment:
nit: we don't really need to use a map here, or loop through all the percentages.

We can just keep track of the last percent output, and compare against it each time:
```
pct := (100 * received) / int64(deal.Transfer.Size)
outputPct := pct % 10
if outputPct != lastOutputPct {
    lastOutputPct = outputPct
    log(outputPct)
}
```

Reply:
Good idea

Reply:
This dosen't work because it logs the output every one percent. We wanna log only every X percent so we will have to use a map.